pkg/apis/core/validation/validation.go

```diff
@@ -4042,6 +4045,20 @@ func ValidateNodeUpdate(node, oldNode *core.Node) field.ErrorList {
 	return allErrs
 }
 
+func validateNodeConfigSource(source *core.NodeConfigSource, fldPath *field.Path) field.ErrorList {
+	count := int(0)
+	if source.ConfigMapRef != nil {
+		count++
+	}
+	// add more subfields here in the future as they are added to NodeConfigSource
+
+	// exactly one reference subfield must be non-nil
+	if count != 1 {
+		return field.ErrorList{field.Invalid(fldPath, source, "exactly one reference subfield must be non-nil")}
+	}
+	return field.ErrorList{}
```

Comment:
nit: convention in validation functions is to create an empty allErrs at the start of a function, append validation errors to it, and return it at the end

Reply:
done